File: haskell/string/1168.hs
-- title: LED
-- number of the problem: 1168
-- date: 24/09/20
-- author: ltsdw

numLeds :: String -> Int
numLeds (x:xs)
    | null xs  = if x == '1' then 2 else if x == '2' || x == '5' || x == '3' then 5 else if x == '4' then 4 else if x == '7' then 3 else if x == '8' then 7 else 6
    | x == '1'  = 2 + numLeds xs
    | x == '2' || x == '5' || x == '3'  = 5 + numLeds xs
    | x == '4'                          = 4 + numLeds xs
    | x == '7'                          = 3 + numLeds xs
    | x == '8'                          = 7 + numLeds xs
    | otherwise                         = 6 + numLeds xs

qtdLeds :: Int -> IO ()
qtdLeds n_cases = do
    if n_cases > 0
        then do
            leds <- getLine
            putStrLn (show (numLeds leds) ++ " " ++ "leds")
            qtdLeds (n_cases - 1)
    else do
        return ()

main :: IO ()
main = do
    l1 <- getLine
    let n_cases = read l1 :: Int
    qtdLeds n_cases

File: c/string/1168.c
/* title: LED
 * problem number: 1168
 * date: 17/09/20
 * author: ltsdw
 */

#include <stdio.h>
#include <string.h>

int check_leds(char *leds);

void printNumLeds(int num_leds);

int main(void) {
    int n_cases;
    char leds[102];

    scanf("%d", &n_cases);

    for (int i = 0; i < n_cases; i++) {
        scanf("%s", leds);

        printNumLeds(check_leds(leds));
    }

    return 0;
}

int check_leds(char *leds) {
    int result = 0;
    int len = strlen(leds);

    for (int i = 0; i < len; i++) {
        if (leds[i] == '1') {
            result += 2;
        }
        if (leds[i] == '2') {
            result += 5;
        }
        if (leds[i] == '3') {
            result += 5;
        }
        if (leds[i] == '4') {
            result += 4;
        }
        if (leds[i] == '5') {
            result += 5;
        }
        if (leds[i] == '6') {
            result += 6;
        }
        if (leds[i] == '7') {
            result += 3;
        }
        if (leds[i] == '8') {
            result += 7;
        }
        if (leds[i] == '9') {
            result += 6;
        }
        if (leds[i] == '0') {
            result += 6;
        }
    }

    return result;
}

void printNumLeds(int num_leds) {
    printf("%d leds\n", num_leds);

    return;
}
